src/components/hoc/RestrictedRoute.js

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { Route, Redirect } from 'react-router';
+
+export default function RestrictedRoute({
+  isAllowed = false,
+  component: Component,
+  render: RenderedComponent,
+  redirectTo,
+  ...routeProps
+}) {
+  return (
+    <Route
+      {...routeProps}
+      render={(props) => {
+        if (isAllowed) {
+          // Compatibility with both Route render or Route component
+          if (!!RenderedComponent) {
+            return RenderedComponent(props);
+          }
+          return <Component {...props} />
```

Comment:
What is the difference between 18 and 20? Can you not combine them into one line without the if/else?

Reply:
<RenderedComponent {...props}/>

Reply:
No, this allows the component to use both the `render` and `component` props like how React Router does normally. If `render` is used then `component` will be undefined and vice-versa.